File: workstation/engine/src/main/java/io/xj/engine/audio/AudioInMemory.java
package io.xj.engine.audio;

import io.xj.hub.pojos.InstrumentAudio;

import javax.sound.sampled.AudioFormat;
import java.util.UUID;

/**
 * Represents an audio file loaded into memory.@param id
 *
 * @param format          the audio format of the audio file
 * @param pathToAudioFile the path to the audio file
 * @param data            2D array of floats representing the audio file-- the first channel is the sample index, the second channel is the channel index
 */
public record AudioInMemory(InstrumentAudio audio, AudioFormat format, String pathToAudioFile, float[][] data) {
  public UUID getId() {
    return audio.getId();
  }

  public String getWaveformKey() {
    return audio.getWaveformKey();
  }

  public boolean isDifferent(InstrumentAudio other) {
    return !audio.getId().equals(other.getId()) || !audio.getWaveformKey().equals(other.getWaveformKey());
  }
}

File: engine/src/audio/AudioInMemory.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/audio/AudioInMemory.h"

using namespace XJ;

UUID AudioInMemory::getId() {
  return audio.id;
}

std::string AudioInMemory::getWaveformKey() {
  return audio.waveformKey;
}

bool AudioInMemory::isDifferent(InstrumentAudio other) {
  return audio.id != other.id || audio.waveformKey != other.waveformKey;
}
